### DISCUSSION TITLE
|filtron| cannot create user data directory: /usr/local/filtron/snap/go/9605: Permission denied

### DISCUSSION THREAD

**Discussion by MachSpeed6** [searx/searx#3217] [Installation issue: Official wiki]
Ubuntu 20.04.4, installing with Apache

Relevant Wiki articles: 
http://searx.github.io/searx/utils/filtron.sh.html#filtron-sh
http://searx.github.io/searx/admin/installation-apache.html

After setting up SearX initially, I decided to follow the guide's advice and set up Filtron and Morty as well as HTTPS encryption.
However, I ran into some bumps installing Filtron. No matter what I did I recieved this error every time
```|filtron| cannot create user data directory: /usr/local/filtron/snap/go/9605: Permission denied```
I ran the install script:
```sudo -H ./utils/filtron.sh install all```
At the end I opted for debugging and the only hint it gave me was 
```systemctl enable  filtron.service
systemctl restart filtron.service
systemctl status --no-pager filtron.service
● filtron.service - filtron
     Loaded: loaded (/lib/systemd/system/filtron.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2022-04-28 21:11:10 CDT; 25ms ago
   Main PID: 1250387 ((filtron))
      Tasks: 1 (limit: 4506)
     Memory: 336.0K
     CGroup: /system.slice/filtron.service

Apr 28 21:11:10 server-HP-PC systemd[1]: Started filtron.
Apr 28 21:11:10 server-HP-PC systemd[1250387]: filtron.service: Failed to execute command: No such file or directory
Apr 28 21:11:10 server-HP-PC systemd[1250387]: filtron.service: Failed at step EXEC spawning /usr/local/filtron/go-apps/bin/filtron: No … directory
Apr 28 21:11:10 server-HP-PC systemd[1]: filtron.service: Main process exited, code=exited, status=203/EXEC
Apr 28 21:11:10 server-HP-PC systemd[1]: filtron.service: Failed with result 'exit-code'.
Hint: Some lines were ellipsized, use -l to show in full.
** press any [KEY] to continue **

ERROR: Filtron is not listening on: http://127.0.0.1:4004
INFO:  Apache is installed on this host.
```

I tried changing the owner, changing the folder permissions, changing the root folder permissions, and running as root
Even turning it off and back on didn't solve the issue.

I suspect it's an issue with Go or Snap, but I'm not sure.

**Comment by ZhenyaPav** [comment]
Having the same issue. Did you find a solution?

**Reply by coder0987** [reply]
Sadly no. I believe it had something to do with an incorrectly configured part of the snap package, so maybe try using the SearxNG docker image?
